libflux/flux-core/src/bin/fluxdoc.rs

```diff
@@ -209,24 +213,93 @@ fn lint(
         pass = false;
     }
     // Evaluate doc examples
-    let mut executor = CLIExecutor {
+    let executor = CLIExecutor {
         path: flux_cmd_path,
     };
-    for d in docs.iter_mut() {
+
+    let tests = docs.par_iter_mut().enumerate().map(|(i, d)| {
         if !exceptions.contains(&d.path.as_str()) {
-            match example::evaluate_package_examples(d, &mut executor) {
-                Ok(_) => {}
+            (i, example::evaluate_package_examples(d, &executor))
+        } else {
+            (i, Vec::new())
+        }
+    });
+
+    let mut test_count = 0;
+    consume_sequentially(tests, |results| {
+        for result in results {
+            test_count += 1;
+            match result {
+                Ok(name) => eprintln!("OK ... {}", name),
                 Err(e) => {
-                    println!("Error {:?}\n", e);
+                    eprintln!("Error {:?}\n", e);
                     pass = false;
                 }
             }
         }
+    });
+
+    if pass {
+        eprintln!("Finished running {} tests", test_count);
+        Ok(())
+    } else {
+        bail!("docs do not pass lint")
+    }
+}
+
+/// Iterates through `iter` in parallel however each item is passed to `consume` in the same order
+/// that they were produced by the iterator as long as the iterator uses `enumerate` to supply
+/// indices. Will deadlock if the indicies passed are not a complete 0..N sequence.
+fn consume_sequentially<T>(
+    iter: impl IndexedParallelIterator<Item = (usize, T)>,
+    mut consume: impl FnMut(T) + Send,
+) where
+    T: Send,
+{
+    let (sender, receiver) = std::sync::mpsc::sync_channel::<Element<T>>(12);
```

Comment:
Curious about the channel bounds here - what was `12` based on?

Reply:
https://xkcd.com/221/

Reply:
Ah. :sweat_smile: